### DISCUSSION TITLE
"Store Heap" - Dynamically allocated e-commerce stores !

### DISCUSSION THREAD

**Discussion by Jd3s** [scaleracademy/open-source-projects#67] [Ideas]
Store _Heap_ - _Dynamically allocated_ stores. Hope y'all got the reference. 😌

_**StoreHeap**_ serves as a platform upon which anyone can build their online e-commerce stores. Essentially, one big factory design pattern for bringing local shops online. 

**_The Why ?_**

- The grocery shop owner who takes orders via WhatsApp messages, sends bills back to you, receives money via _GPay_, and gets the order delivered to your house. 🛍️
- The restaurant guy in a small town, who takes orders via call, receives money via _paytm_ and prepares a takeaway/home delivery for you. 🍔
- The fashion influencer on Instagram who takes orders via DMs and accepts money via _phonepe_ and sends your order via courier. 👠

What do they have in common,

- They all somehow let their customer know the products they sell, take orders, accept payment. 
- Hiring a developer to build a site/app for them, buying a domain name, hosting the site, integrating a payment gateway is a hustle for them.  💻 
- Being online, helps increase their sales and reduces the efforts for their customers. Especially, during this pandemic. 

_Enters Store Heap,_ 

1.Shop owners sign up for their free stores.
2. Pick a name, color, upload their logo, upload products, their prices. ( Optional - images for the products )
3. Voila. They get a URL, which they can share with their customers and take orders and the integrated payment gateway helps the shop owners to receive payments as well.

_Thank thy abstraction_. 🐱‍💻
 All thse URLs generated for each shop are along the pattern of, <Our domain name> / <the shop name>. Eg: StoreHeap/MomoCorner. 

The application servers, domain name, the databases are essentially the same for all the shops. There's probably a "fk_shop_id" column in product, order, cart tables that maps to "shop_id" column in the shops table. (I hope you get the gist. Can explain if you have questions.) 
To the shopkeeper and his customers, the URL feels like their own digital website. 

Features :
Payments gateway integration - RazorPay / PayU
Admin panel - where shop keepers can change their product catalog, price, availability of the product etc,
Telegram - So that the shopkeepers can receive the orders in their phone, when someone orders from their site.
Telegram bots - Wherein the shopkeeper can update the order status (Packed / delivered etc.,) in the chat where he received the order message.

**Comment by shikharraje** [comment]
How is this different from Shopify or Bikayi?

**Reply by championswimmer** [reply]
While it is good concept to build upon for sure, but yes, what's the major difference from mydukaan.io or Shopify

**Comment by johnwill0** [comment]
I am working on a clothing brand named [clothing manufacturers los angeles]([url](url)), we are providing its services in los angeles only. No we are trying to expand it globally. I will try your cms and will give my honest feedback about it.

**Comment by Allana589** [comment]
We are working on the pet site that are related to [can dogs eat freeze dried mango?](<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Ayesha/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Ayesha/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<div data-ui-name=Flex role=cell headers=igc-table-ui-kit-8819-phrase name=phrase
style='border-bottom:var(--intergalactic-border-secondary, #e0e1e9);border-image: initial;
font-variant-ligatures: normal;font-variant-caps: normal;font-variant-numeric: inherit;
font-variant-east-asian: inherit;font-variant-alternates: inherit;font-stretch: inherit;
font-size:var(--intergalactic-fs-200, 14px);line-height:var(--intergalactic-lh-200, 142%);
font-optical-sizing: inherit;font-kerning: inherit;font-feature-settings: inherit;
font-variation-settings: inherit;vertical-align:initial;display:flex;
flex: 1 1 auto;box-sizing: border-box;min-height: 45px;background-color:var(--intergalactic-bg-primary-neutral, #ffffff);
position:sticky;orphans: 2;text-align:start;widows: 2;-webkit-text-stroke-width: 0px;
text-decoration-thickness: initial;text-decoration-style: initial;text-decoration-color: initial;
width:var(--phrase_width);left:var(--checkbox_width);justify-content: flex-start'>

<div data-ui-name=Box style='vertical-align:initial;margin-right:0px;
padding-bottom:0px;padding-top:0px'>

<div data-ui-name=Flex style='vertical-align:initial;display:flex;margin-right:
0px;padding-bottom:0px;padding-top:0px'>

<div style='vertical-align:initial;white-space:break-spaces'><span data-ui-name=Link.Text
style='border-top-style:initial;border-right-style:initial;border-left-style:
initial;border-color:transparent;border-image: initial;vertical-align:initial;
transition: border-bottom-color 0.15s ease-in-out 0s'>

<div data-ui-name=Flex style='vertical-align:initial;display:inline-flex'>

<div data-ui-name=Flex style='vertical-align:initial;display:inline-flex'>

<div style='vertical-align:initial;margin-right:0px;padding-bottom:0px;
padding-top:0px'></div>

</div>

</div>

</div>

</div>

</div>

</div>

</div>

<div data-ui-name=Flex role=cell headers=igc-table-ui-kit-8819-intents name=intents
style='border-bottom:var(--intergalactic-border-secondary, #e0e1e9);border-image: initial;
font-variant-ligatures: normal;font-variant-caps: normal;font-variant-numeric: inherit;
font-variant-east-asian: inherit;font-variant-alternates: inherit;font-stretch: inherit;
font-size:var(--intergalactic-fs-200, 14px);line-height:var(--intergalactic-lh-200, 142%);
font-optical-sizing: inherit;font-kerning: inherit;font-feature-settings: inherit;
font-variation-settings: inherit;vertical-align:initial;display:flex;
flex: 1 1 auto;box-sizing: border-box;min-height: 45px;background-color:var(--intergalactic-bg-primary-neutral, #ffffff);
orphans: 2;text-align:start;widows: 2;-webkit-text-stroke-width: 0px;
text-decoration-thickness: initial;text-decoration-style: initial;text-decoration-color: initial;
width:var(--intents_width);justify-content: center'>

<div data-ui-name=Box style='vertical-align:initial;margin-right:0px;
padding-bottom:0px;padding-top:0px'>

<div data-ui-name=Flex style='vertical-align:initial;align-items: flex-start;
display:flex'>

<div data-ui-name=Box style='vertical-align:initial;margin-right:0px;
padding-bottom:0px;padding-top:0px'>

<div data-ui-name=Flex style='vertical-align:initial;display:flex;justify-content: flex-end'>

<div data-ui-name=Tooltip aria-haspopup=false style='vertical-align:initial;
display:flex;margin-right:0px;padding-bottom:0px;padding-top:0px'>

<div data-ui-name=Flex data-at=intent-badge-informational style='vertical-align:
initial;display:flex;margin-right:0px;padding-bottom:0px;padding-top:0px'>

<div data-ui-name=Flex style='vertical-align:initial;border-radius: 4px;
cursor:pointer !important;display:flex;align-items: center;justify-content: center'></div>

</div>

</div>

</div>

</div>

</div>

</div>

</div>

</span>

<div data-ui-name=Box data-at=ko-icon style='vertical-align:initial;display:
inline-block'>

<body link="#0563C1" vlink="#954F72">



https://petarenapro.com/can-dogs-eat-dry-fruits/
--




</body>

</html>
) Now we are trying to expand this. So give your honest feedback about it.

**Comment by duncan324** [comment]
In this ecom business, delivery is an issue. To sort out this issue I am using <a href="https://thetcstracking.pk/">TCS Tracking </a> to delivery all kind of packages. You can give it a try.

**Comment by duncan324** [comment]
We are working on [TCS Shipment tracking](https://thetcstracking.pk/). We will try your cms for sure and give our feedback.

**Comment by Allana589** [comment]
Love the concept of 'Store Heap' for dynamically allocated e-commerce stores! A great way to optimize and manage inventory. Also, a quick reminder for pet owners: while browsing for your favorite items, remember to keep an eye on what's safe for your furry friends - important to know if items like <a href=
"https://petarenapro.com/can-dogs-drink-coffee/">will coffee hurt dogs</a> to ensure a pet-friendly shopping experience.

**Comment by Allana589** [comment]
"Absolutely intrigued by Store Heap's concept of dynamically allocated e-commerce stores! Speaking of safe options, have you ever explored if <a href="https://petarenapro.com/can-dog-eat-papaya/">papaya safe for dogs</a>

**Comment by Allana589** [comment]
Love the concept of 'Store Heap'—sounds like a game-changer for e-commerce! 🛒💡 Curious, though—any tips on <a href="https://petarenapro.com/how-to-adjust-invisible-fence-collar-strength/">how to adjust Invisible Fence brand collar strength?</a>

**Comment by Allana589** [comment]
Exciting news about Store Heap's dynamically allocated e-commerce stores! 🛒💻 By the way, while we're on the topic of variety, did you know that <a href="https://petarenapro.com/can-dogs-eat-carrots/">can dogs eat yellow carrots?</a>They're a safe and healthy treat for our furry friends.

**Comment by Allana589** [comment]
Exciting stuff with Store Heap's dynamically allocated e-commerce stores! Innovation is key in this space. By the way, when diving into new features, it's good to keep in mind pet-friendly celebrations too. Speaking of which, <a href="https://petarenapro.com/can-dogs-eat-cake/">can dogs eat birthday cake?</a> It's always a fascinating topic to ensure our furry companions stay safe during our festivities. Cheers to innovation and pet safety
